git/index/util.py

```diff
@@ -40,7 +40,8 @@ class TemporaryFileSwap(object):
 
     def __init__(self, file_path: PathLike) -> None:
         self.file_path = file_path
-        self.tmp_file_path = str(self.file_path) + tempfile.mktemp("", "", "")
+        self.tmp_file_path = str(self.file_path) + osp.basename(tempfile.mkstemp("", "", "")[1])
```

Comment:
The user is responsible for closing the file in `[0]` or else it will definitely be leaked.